common/models/user.js

```diff
@@ -993,18 +995,22 @@ module.exports = function(User) {
 
   User.validatePassword = function(plain) {
     var err;
-    if (plain && typeof plain === 'string' && plain.length <= MAX_PASSWORD_LENGTH) {
-      return true;
+    if (!plain || typeof plain !== 'string') {
+      err = new Error(g.f('Invalid password.'));
+      err.code = 'INVALID_PASSWORD';
```

Comment:
@STRML @bajtos 
I agree about removing the print out of the password and unifying the error code as well, however, here is a thought, I think we should not even point to the direction of the error .. In other words, we should not let the error code say if it is wrong username and/or password .. It should be more vague for security purposes .. Like 
`INVALID_LOGIN_INFO` or something similar.. That's my 2 cents .. the rest of the PR is LGTM